### PR TITLE
Disallow unknown exports

### LINKED ISSUE #7845: Reserve `config` export
### Describe the problem

We've alluded to a vague plan to support route-level deployment config in the future — something like this...

```js
export const config = {
  runtime: 'edge'
};
```

...where `config` is a freeform (but serializable) object that gets passed to adapters so that they can use it to customise the deployment.

At the moment, you can export anything from a `+(layout|page)(.server)?.js` file. This means that adding a new _known_ export would be a breaking change, as someone might already be doing `export const config = {...}` for some other reason.

### Describe the proposed solution

Reserve the `config` name, i.e. throw an error if a route file exports a binding called `config`.

### Alternatives considered

Prevent _any_ unknown exports. This is safer, but might feel unduly restrictive.

### Importance

would make my life easier

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
Closes #7845, by disallowing all unknown exports. For a minority of apps, this will be a breaking change, which is solved easily by moving the offending exports into a non-route file and importing them.

A related breaking change is that prerendering will fail (whether or not you're actively using prerendering) if modules cannot be imported for analysis at build time (e.g. because a file references a missing variable or something). This is a good thing, and something we should have done before, since it prevents surprising runtime errors.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: .changeset/friendly-jobs-perform.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+[breaking] disallow unknown exports
```
Comment (Conduitry) [inline .changeset/friendly-jobs-perform.md:5]: This should be clearer about where unknown exports are disallowed. Is it all `+*.js` files?

Comment (Rich-Harris) [reply .changeset/friendly-jobs-perform.md:5]:
```suggestion
[breaking] disallow unknown exports from `+(layout|page)(.server)?.js` and `+server.js` files
```

File: .changeset/friendly-jobs-perform.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+[breaking] disallow unknown exports from `+(layout|page)(.server)?.js` and `+server.js` files
```
Comment (Conduitry) [inline .changeset/friendly-jobs-perform.md:5]: Do we now want to call out the `_` exception in the changeset?

Either way, now that there's this exception, it should definitely be documented somewhere, and perhaps we could also make the error message nicer. ('If you're exposing this to facilitate testing, consider doing...')

Comment (dummdidumm) [inline .changeset/friendly-jobs-perform.md:5]:
```suggestion
[breaking] disallow unknown exports (except when starting with an underscore) from `+(layout|page)(.server)?.js` and `+server.js` files
```

**Review by dummdidumm — APPROVED** [review]

**Pull request merged**